Sources/CodableKit/DecoderUnwrapper.swift

```diff
@@ -0,0 +1,14 @@
+/// Used to unwrap the `Decoder` from a private implementation like `JSONDecoder`.
+///
+///     let unwrapper = try JSONDecoder().decode(DecoderUnwrapper.self, from: ...)
+///     print(unwrapper.decoder) // Decoder
+///
+public struct DecoderUnwrapper: Decodable {
```

Comment:
Cool, but is this something we should really be doing in practice?

Reply:
It's incredibly useful for getting raw access to Decoder methods like `container(keyedBy:)`, `singleValueContainer()`, etc. I find myself re-implementing this in most frameworks that use `Codable`.

Reply:
👍 Then I think we should add a convenience method to get internal decoders from `DataDecoder` types.

Reply:
```swift
JSONDecoder.decoder(with: data) // -> _JSONDecoder
```

Reply:
Agreed, just put a comment on #2.